pkg/subctl/cmd/validate_fw_vxlan.go

```diff
@@ -59,47 +62,52 @@ func validateFirewallVxLANConfig(cmd *cobra.Command, args []string) {
 		}
 
 		status.End(cli.Success)
-		validateVxLANConfigWithinCluster(item.config, item.clusterName, submariner)
+		validationStatus = validationStatus && validateVxLANConfigWithinCluster(item.config, item.clusterName, submariner)
+	}
+
+	if !validationStatus {
+		os.Exit(1)
 	}
 }
 
-func validateVxLANConfigWithinCluster(config *rest.Config, clusterName string, submariner *v1alpha1.Submariner) {
+func validateVxLANConfigWithinCluster(config *rest.Config, clusterName string, submariner *v1alpha1.Submariner) bool {
 	status.Start(fmt.Sprintf("Checking the firewall configuration to determine if VXLAN traffic is allowed"+
 		" in cluster %q", clusterName))
-	validateFWConfigWithinCluster(config, submariner)
+	validationStatus := validateFWConfigWithinCluster(config, submariner)
 	status.End(status.ResultFromMessages())
+	return validationStatus
 }
 
-func validateFWConfigWithinCluster(config *rest.Config, submariner *v1alpha1.Submariner) {
+func validateFWConfigWithinCluster(config *rest.Config, submariner *v1alpha1.Submariner) bool {
 	if submariner.Status.NetworkPlugin == "OVNKubernetes" {
 		status.QueueSuccessMessage("This check is not necessary for the OVNKubernetes CNI plugin")
-		return
+		return true
 	}
 
 	clientSet, err := kubernetes.NewForConfig(config)
 	if err != nil {
 		message := fmt.Sprintf("Error creating API server client: %s", err)
 		status.QueueFailureMessage(message)
-		return
+		return false
 	}
 
 	gateways := getGatewaysResource(config)
 	if gateways == nil {
 		status.QueueWarningMessage("There are no gateways detected on the cluster.")
-		return
+		return false
 	}
 
-	if len(gateways.Items[0].Status.Connections) == 0 {
+	if len(gateways.Items) == 0 || len(gateways.Items[0].Status.Connections) == 0 {
```

Comment:
```suggestion
	if len(gateways.Items[0].Status.Connections) == 0 {
```

Reply:
Ah, ditto to the review I was just working on.